qkernel/src/lib.rs

```diff
@@ -332,6 +330,113 @@ pub extern "C" fn syscall_handler(
     }
 }
 
+// syscall_handler implementation for aarch64:
+// Unlike x86, this function is NOT directly called 
+// from the asm code (vector). The C calling convention
+// is not necessary here.
+// TODO move this function to a proper place.
+#[no_mangle]
+#[cfg(target_arch = "aarch64")]
+pub fn syscall_dispatch_aarch64(
```

Comment:
Explanation:

I think this "dispatch" function is perhaps defined in a wrong place, but I'm trying to keep the flow similar to the x86 version (the `syscall_handler` function defined in lib.rs).